File: mqtt-old/src/main/java/org/sheedon/mqtt/ResponseBody.java
package org.sheedon.mqtt;

import org.eclipse.paho.client.mqttv3.MqttMessage;

/**
 * 反馈内容
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2020/2/26 16:31
 */
public class ResponseBody {


    private String data;
    private int qos = 1;
    private boolean retained = false;
    private boolean dup = false;
    private int messageId;

    public void updateMqttMessage(MqttMessage message, String charset) {
        this.qos = message.getQos();
        this.retained = message.isRetained();
        this.dup = message.isDuplicate();
        this.messageId = message.getId();
        this.data = CharsetUtils.changeCharset(message.getPayload(), charset);
    }

    public String getBody() {
        return data;
    }

    public void updateBody(String data) {
        this.data = data;
    }

    public void close() {

    }
}

File: mqtt/src/main/java/org/sheedon/mqtt/ResponseBody.kt
package org.sheedon.mqtt

import org.eclipse.paho.client.mqttv3.MqttMessage

/**
 * 反馈响应消息体
 * 包含内容，「订阅主题」+「消息内容」+「消息质量」（客户端无用）+「是否保留」（客户端无用）
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/30 11:13 上午
 */
class ResponseBody internal constructor() : MqttMessage() {

    var topic: String? = null
        private set
    var data: String? = null
        get() {
            if (field != null) {
                return field
            }

            field = super.getPayload().toString()
            return field
        }

    constructor(topic: String, mqttMessage: MqttMessage) : this() {
        this.topic = topic
        super.setPayload(mqttMessage.payload)
        super.setQos(mqttMessage.qos)
    }


}